tools/issue-labeler/src/IssueLabelerService/AzureSdkIssueLabelerService.cs

```diff
@@ -69,32 +69,48 @@ public async Task<ActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "P
                 return EmptyResult;
             }
 
-            try{
+            if(bool.Parse(config.EnableAnswers))
+            {
+                try{
 
-                // Get the Qna model based on configuration
-                var qnaService = _answerServices.GetAnswerService(config);
+                    // Get the Qna model based on configuration
+                    var qnaService = _answerServices.GetAnswerService(config);
 
-                var answer = await qnaService.AnswerQuery(issue, labels);
+                    var answer = await qnaService.AnswerQuery(issue, labels);
 
-                TriageOutput result = new TriageOutput
+                    TriageOutput result = new TriageOutput
+                    {
+                        Labels = labels.Values,
+                        Answer = answer.Answer,
+                        AnswerType = answer.AnswerType
+                    };
+
+                    return new JsonResult(result);
+                }
+                catch (Exception ex)
                 {
-                    Labels = labels.Values,
-                    Answer = answer.Answer,
-                    AnswerType = answer.AnswerType
-                };
+                    _logger.LogError($"Error commenting on issue #{issue.IssueNumber} in repository {issue.RepositoryName}: {ex.Message}{Environment.NewLine}\t{ex}{Environment.NewLine}");
 
-                return new JsonResult(result);
+                    TriageOutput result = new TriageOutput
```

Comment:
We can avoid the duplication here by moving things around a bit.  Something like:

```csharp

var result = new TriageOutput { Labels = labels.Values };

if (iWantAnswersNao)
{
    try
    {
        var answers = GetAnswers(...);
        result.Answer = answers.Answer;
        result.AnswerType = answers.AnswerType;
    }
    catch (Exception ex)
    {
        // Log - ERROR, OMG OMG!

        // Very unlikely this is needed, but covers any weird corner case.
        result.Answer = null;
        result.AnswerType = null;
    }

    return new JsonResult(result);
}
```